WordPressAuthenticator/NUX/Button/NUXStackedButtonsViewController.swift

```diff
@@ -0,0 +1,227 @@
+        dividerLabel.textColor = color
+        dividerLabel.text = NSLocalizedString("Or", comment: "Divider on initial auth view separating auth options.").localizedUppercase
+    }
+
+    func updateShadowViewEdgeConstraints() {
+        guard let layoutGuide = shadowLayoutGuide,
+              let shadowView = shadowView else {
+            return
+        }
+
+        NSLayoutConstraint.deactivate(shadowViewEdgeConstraints)
+        shadowView.translatesAutoresizingMaskIntoConstraints = false
+
+        shadowViewEdgeConstraints = [
+            layoutGuide.leadingAnchor.constraint(equalTo: shadowView.leadingAnchor),
+            layoutGuide.trailingAnchor.constraint(equalTo: shadowView.trailingAnchor),
+        ]
+
+        NSLayoutConstraint.activate(shadowViewEdgeConstraints)
+    }
+
+    // MARK: - Dynamic type
+    func didChangePreferredContentSize() {
+        reloadViews()
+    }
+}
+
+extension NUXStackedButtonsViewController {
+    override open func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+        super.traitCollectionDidChange(previousTraitCollection)
+        if previousTraitCollection?.preferredContentSizeCategory != traitCollection.preferredContentSizeCategory {
+            didChangePreferredContentSize()
+        }
+    }
+}
+
+extension NUXStackedButtonsViewController {
+
+    /// Sets the parentViewControlleras the receiver instance's container. Plus: the containerView will also get the receiver's
+    /// view, attached to it's edges. This is effectively analog to using an Embed Segue with the NUXButtonViewController.
+    ///
+    public func move(to parentViewController: UIViewController, into containerView: UIView) {
+        containerView.translatesAutoresizingMaskIntoConstraints = false
+        containerView.addSubview(view)
+        containerView.pinSubviewToAllEdges(view)
+
+        willMove(toParent: parentViewController)
+        parentViewController.addChild(self)
+        didMove(toParent: parentViewController)
+    }
+
+    /// Returns a new NUXButtonViewController Instance
+    ///
+    public class func instance() -> NUXStackedButtonsViewController {
+        let storyboard = UIStoryboard(name: "NUXButtonView", bundle: WordPressAuthenticator.bundle)
+        guard let buttonViewController = storyboard.instantiateViewController(withIdentifier: "NUXStackedButtonsViewController") as? NUXStackedButtonsViewController else {
+            fatalError()
+        }
+
+        return buttonViewController
+    }
+}
+
+private extension NUXButton {
+    func configure(withConfig buttonConfig: NUXButtonConfig?, and style: NUXButtonStyle?) {
+        guard let buttonConfig = buttonConfig else {
+            isHidden = true
+            return
+        }
+
+        if let attributedTitle = buttonConfig.attributedTitle {
+            setAttributedTitle(attributedTitle, for: .normal)
+            socialService = buttonConfig.socialService
```

Comment:
❓  I'm curious: why does `socialService`only get set in this case?

Reply:
Looks like it was because `socialService` is non-nil when a button has `attributedTitle`.

Changed in https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/695/commits/c83a8c9f28660dd34f1c344d3149b73dd70f0a67